doc/reference/plutus/babbage-script-example.md

```diff
@@ -4,20 +4,26 @@
 
 A reference script is a script that exists at a particular transaction output. It can be used to witness, for example, a UTxO at the corresponding script address of said reference script. This is useful because the script does not have to be included in the transaction anymore, which significantly reduces the transaction size.
 
+## What is an inline datum?
+
+An inline datum, is a datum that exists at a transaction output. We no longer have to include a datum within our transaction for our plutus spending scripts. Instead we can specify the transaction output where our datum exists to be used in conjunction with our Plutus spending script. This reduces the overall size of our transaction.
+
 ### An example of using a Plutus V2 reference script
 
-Below is an example that shows how to use a Plutus spending script. Here we discuss a [shell script example of how to use a reference script to spend a tx input](scripts/plutus/example-reference-script-usage.sh). This is a step-by-step process involving:
+Below is an example that shows how to use a Plutus spending script and an inline datum. Here we discuss a [shell script example of how to use a reference script to spend a tx input](scripts/plutus/example-babbage-script-usage.sh). This is a step-by-step process involving:
 
 + the creation of the `Required Redeemer` Plutus txin script
 + the creation of the `Required Redeemer` Plutus script at a transaction output (creation of the reference script)
-+ sending ada and a datum to the Plutus script address
++ the creation of the inline datum at a transaction output
++ sending ada to the Plutus script address
 + spending ada at the Plutus script address using the Plutus reference script
 
 In this example we will use the [Required Redeemer](scripts/plutus/scripts/v2/required-redeemer.plutus) Plutus spending script. In order to execute a reference Plutus spending script, we require the following:
 
 - Collateral tx input(s) - these are provided and are forfeited in the event the Plutus script fails to execute.
-- A Plutus tx output with accompanying datum hash. This is the tx output that sits at the Plutus script address. It must have a datum hash, otherwise, it is unspendable.
+- A Plutus tx output. This is the tx output that sits at the Plutus script address. Note because we are using an inline datum, it is not mandatory that the utxo at the plutus script address has a datum hash.
```

Comment:
We should stress this point, how about: "Note because we are using an inline datum, the utxo at the plutus script address must not have a datum hash because datum witnessing requirements cannot be satisfied with reference inputs. Outputs with datum hashes must have datum included in transaction body as witness."

Reply:
>Actually, it strictly cannot have a datum hash because the inline datum we reference cannot witness it.

Yep this was a separate point I was making in light of inline datums. But you are right I should make this clearer.

Reply:
I realized from your comment I need to add the additional parser I was talking about (including a reference input that is not witnessing anything)

Reply:
> Actually, it strictly cannot have a datum hash because the inline datum we reference cannot witness it. It would then need the datum to be attached to the tx, which means we cannot also reference the inline datum!

Yep my wording is poor. "it is not mandatory that the utxo at the plutus script address has a datum hash" what I mean here is you can use an inline datum in stead of a datum hash.